lib/fog/aws/models/rds/server.rb

```diff
@@ -104,16 +104,22 @@ def modify(immediately, options)
         end
 
         def save
-          requires :engine
-          requires :allocated_storage
-          requires :master_username
-          requires :password
-
-          self.flavor_id ||= 'db.m1.small'
-
-          data = service.create_db_instance(id, attributes_to_params)
-          merge_attributes(data.body['CreateDBInstanceResult']['DBInstance'])
-          true
+          if source_snapshot_id
```

Comment:
Since this is pretty different, maybe we should just have a separate restore function (instead of having save pull double duty). I suspect that might make it clearer and less error prone for users. What do you think?